### PR TITLE
[Test Proxy] Maps Search migration

### PR DESCRIPTION
**Pull request by vcolin7**
Updated test classes to use Test Proxy and migrated recordings to the assets repo. Also had to modify the base test class to improve `HttpPipeline` creation for test clients.

Had to disable 4 tests that kept failing due to a different service response than what was recorded previously. Here's the issue tracking investigating this and re-enabling said tests: #35979

### CODE REVIEW

File: sdk/maps/azure-maps-search/pom.xml [hunk 1]
```diff
@@ -109,5 +114,11 @@
       <version>3.4.30</version> <!-- {x-version-update;io.projectreactor:reactor-test;external_dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core</artifactId>
+      <version>1.42.0-beta.1</version>
+      <scope>compile</scope>
+    </dependency>
```
Comment (samvaity) [inline sdk/maps/azure-maps-search/pom.xml:122]: remove?

File: sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java [hunk 2]
```diff
@@ -56,59 +70,62 @@ protected void beforeTest() {
...
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode() || interceptorManager.isPlaybackMode()) {
+            interceptorManager.addSanitizers(
+                Collections.singletonList(
+                    new TestProxySanitizer("subscription-key", ".+", "REDACTED", TestProxySanitizerType.HEADER)));
```
Comment (samvaity) [inline sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java:92]: nit: why not just use the core-test unrelased dependency?

File: sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java [hunk 3]
```diff
@@ -56,59 +70,62 @@ protected void beforeTest() {
...
         return builder;
     }
 
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode() || interceptorManager.isPlaybackMode()) {
```
Comment (samvaity) [inline sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java:89]: nit
```suggestion
        if (!interceptorManager.isLiveMode()) {
```

File: sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java [hunk 4]
```diff
@@ -56,59 +70,62 @@ protected void beforeTest() {
...
+            policies.add(interceptorManager.getRecordPolicy());
+        }
 
-    String getEndpoint() {
-        return interceptorManager.isPlaybackMode()
-                ? "https://localhost:8080"
-                : endpoint;
+        return new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient)
```
Comment (samvaity) [inline sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java:127]: already done on L87

File: sdk/maps/azure-maps-search/pom.xml (hunk 1 above)
Comment (vcolin7) [reply sdk/maps/azure-maps-search/pom.xml:122]: Yeah, I must have added that with IntelliJ by accident, thanks for the catch

File: sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java (hunk 2 above)
Comment (vcolin7) [reply sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java:92]: I will update all dependencies to use the new version once it comes out.

File: sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java (hunk 4 above)
Comment (vcolin7) [reply sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java:127]: Good catch

**Review by joshfree — APPROVED** [review]

**Pull request merged**